File: 2020/old/xgraphics/homeworld2/core/Homeworld2/Graphics/Goblinmesh.cc
#include "Goblinmesh.hh"

namespace Homeworld2
{
    namespace Graphics
    {
        Goblinmesh::Goblinmesh()
        :   x::graphics::reference(),
            singlemesh(),
            __parent(),
            __name()
        {
        }

        Goblinmesh::Goblinmesh(const Homeworld2::Graphics::Goblinmesh & o)
        :   x::graphics::reference(),
            singlemesh(o.singlemesh),
            __parent(o.__parent),
            __name(o.__name)
        {
        }

        Goblinmesh::Goblinmesh(Homeworld2::Graphics::Goblinmesh && o) noexcept
        :   x::graphics::reference(),
            singlemesh(x::move(o.singlemesh)),
            __parent(x::move(o.__parent)),
            __name(x::move(o.__name))
        {
        }

        Goblinmesh::~Goblinmesh()
        {
        }

        Homeworld2::Graphics::Goblinmesh & Goblinmesh::operator=(const Homeworld2::Graphics::Goblinmesh & o)
        {
            if(&o != this)
            {
                singlemesh = o.singlemesh;
                __parent = o.__parent;
                __name = o.__name;
            }
            return *this;
        }

        Homeworld2::Graphics::Goblinmesh & Goblinmesh::operator=(Homeworld2::Graphics::Goblinmesh && o) noexcept
        {
            if(&o != this)
            {
                singlemesh = x::move(o.singlemesh);
                __parent = x::move(o.__parent);
                __name = x::move(o.__name);
            }
            return *this;
        }

        void Goblinmesh::load()
        {
            singlemesh.load();
        }

        void Goblinmesh::unload()
        {
            singlemesh.unload();
        }

        void Goblinmesh::render()
        {
            singlemesh.render();
        }

        const x::string & Goblinmesh::name() const
        {
            return __name;
        }

        Homeworld2::Graphics::Objects::Single * Goblinmesh::build(x::array<x::graphics::texture> & textures, x::array<Material> & materials)
        {
            return new Homeworld2::Graphics::Objects::Single(__name, singlemesh.build(textures, materials));
        }
    }
}

File: 2020/old/discomplexity/x/js/homeworld2/goblinmesh.js
import Obj from '../x/graphics/obj.js';

export default class Goblinmesh extends Obj
{
    constructor(name, parent, status)
    {
        super(name, 0, status);

        this.__singlemesh = null;
        this.__parent = parent;
    }

    get singlemesh(){ return this.__singlemesh; }
    get status(){ return this.__status; }
    get name(){ return this.__name; }
    get parent(){ return this.__parent; }

    set singlemesh(o){ this.__singlemesh = o; }

    set status(v)
    {
        this.__status = v;
        this.__singlemesh.status = v;
    }

    render(gl, model, level)
    {
        this.__singlemesh.render(gl, model, level);

    }

    init(gl)
    {
        this.__singlemesh.init(gl);
    }
}
